src/loaders/lottie/tvgLottieOverrider.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "tvgLottieOverrider.h"
+#include "tvgCommon.h"
+#include "tvgLottieOverrider.h"
+#include "tvgLottieParser.h"
+#include "tvgLottieLoader.h"
+
+using namespace std;
+
+/************************************************************************/
+/* Internal Class Implementation                                        */
+/************************************************************************/
+
+/************************************************************************/
+/* External Class Implementation                                        */
+/************************************************************************/
+
+bool LottieOverrider::overrideSlot(const char* data)
+{
+    auto parser = new LottieParser(strdup(data), this->dirName);
+    parser->comp = this->comp;
+
+    auto sid = parser->getSid();
+    if (!sid) return false;
+
+    auto slots = comp->slots;
+    for (auto s = slots.data; s < slots.end(); ++s) {
+        auto slot = (*s);
+
+        if (!strcmp(slot->sid, sid)) {
+            parser->parse(slot);
+            break;
+        }
+    }
+
+    return true;
+}
+
+
+void LottieOverrider::overrideGradient(LottieObject* target, LottieGradient* gradient)
+{
+    switch (target->type)
+    {
```

Comment:
Please double check the convention:
![image](https://github.com/thorvg/thorvg/assets/3711518/6860c167-b62f-480c-9679-f6d49281627d)